scripts/tasks/version-check.js

```diff
@@ -6,14 +6,45 @@
  */
 
 'use strict';
-const execSync = require('child_process').execSync;
+const execFileSync = require('child_process').execFileSync;
+const yarnCmds = ['yarnpkg', 'yarn'];
 
-function getLocalYarnVersion() {
-  return execSync('yarn --version', {
+function getVersion(cmd) {
+  return execFileSync(cmd, ['--version'], {
     encoding: 'utf-8',
   }).trim();
 }
 
+function getLocalYarnVersion() {
+  let yarnVersion;
+  let caughtError = null;
+
+  const isInstalledYarn = yarnCmds.some(function(cmd) {
+    try {
+      yarnVersion = getVersion(cmd);
+      return true;
+    } catch (err) {
+      caughtError = err;
+      return false;
+    }
```

Comment:
It will try `yarnpkg` first, and then `yarn`. If both of them throw error, and will fail gracefully with a message when error.code is `ENOENT`